lib/commands/device/deactivate.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * @license
+ * Copyright 2016-2020 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { flags } from '@oclif/command';
+import type { IArg } from '@oclif/parser/lib/args';
+import Command from '../../command';
+import * as cf from '../../utils/common-flags';
+import { getBalenaSdk, stripIndent } from '../../utils/lazy';
+
+interface FlagsDef {
+	yes: boolean;
+	help: void;
+}
+
+interface ArgsDef {
+	uuid: string;
+}
+
+export default class DeviceDeactivateCmd extends Command {
+	public static description = stripIndent`
+		Deactivate a device.
+
+		Deactivate a device.
+
+		Note this command asks for confirmation interactively.
+		You can avoid this by passing the \`--yes\` option.
+		`;
+	public static examples = [
+		'$ balena device deactivate 7cf02a6',
+		'$ balena device deactivate 7cf02a6 --yes',
+	];
+
+	public static args: Array<IArg<any>> = [
+		{
+			name: 'uuid',
+			description: 'the UUID of the device to be deactivated',
+			required: true,
+		},
+	];
+
+	public static usage = 'device deactivate <uuid>';
+
+	public static flags: flags.Input<FlagsDef> = {
+		yes: cf.yes,
+		help: cf.help,
+	};
+
+	public static authenticated = true;
+
+	public async run() {
+		const { args: params, flags: options } = this.parse<FlagsDef, ArgsDef>(
+			DeviceDeactivateCmd,
+		);
+
+		const balena = getBalenaSdk();
+		const patterns = await import('../../utils/patterns');
+
+		const uuid = params.uuid;
+		const deactivationWarning = `
+
+Warning! Deactivating a device will charge a fee equivalent to the
+normal monthly cost for the device (e.g. $1 for an essentials device);
+the device will not be charged again until it comes online.
```

Comment:
Grammatically, deactivation will not charge a fee... Balena will charge a fee, because of deactivation. 😀   Perhaps 'incur' instead of 'charge':

```
Warning! Deactivating a device will incur a fee equivalent to the
normal monthly cost for the device (e.g. $1 for an essentials device);
the device will not be charged again until it comes online.
```

Also:
* Re "until it comes online" - are devices charged based on being online/offline, or based or being active/inactive? An active device may be online or offline, and is charged regardless, no? Similarly, an inactive device may be online in the sense of connected to internet (and perhaps still communicating with the user's own servers), and yet not be charged. Is this right? I may be wrong about this, don't assume I know better!
* Re "the device will not be charged" - to be picky, users are charged, not devices (we don't ask devices to pay us money, we ask their owners).

If you also think that these notes are correct, then perhaps:

```
Warning! Deactivating a device will incur a fee equivalent to the
normal monthly cost for the device (e.g. $1 for an essentials device);
device fees will not be charged again until the device is reactivated.
```

Reply:
> until the device is reactivated

By the way, how can the device be reactivated? Do we need another CLI command for this? `balena device activate`? If so, it would make sense for this PR to add both commands (could be in separate commits).

Reply:
@pdcastro yeah we don't have this activate command, but the `until it comes online.` is correct I think, because if it comes online it will get charged.

> An active device may be online or offline, and is charged regardless, no? Similarly, an inactive device may be online in the sense of connected to internet (and perhaps still communicating with the user's own servers), and yet not be charged. Is this right?


this I think is incorrect,

Though these are billing stuff with which I am not familiar, maybe @thgreasi can give us some input here. Also whatever we decide on this message, we should change the corresponding one in the balena-ui in order to match. That's the place I initially took it from.

Reply:
I see! 👍  So a device is reactivated by coming back online, and therefore there is no need for a reactivate command. In this case, do we need to check that the device is offline (well, not online) before deactivating it? What happens if the user deactivates a device that is currently online?

Reply:
@pdcastro the logic you describe I think I've seen it somewhere, while I was searching around to do this feature, so I can try and reproduce on balena-staging to see what errors come out of this. I think the approach should be to have as simple logic as possible, for example, if the API does the necessary checks and just returns a nice error then we shouldn't check the same conditions here in the CLI.

Reply:
Given that there is no such thing as an explicit "reactivate" action (devices are automatically reactivated when they come back online, as I've learned!), I retract my suggestion of using the word "reactivated" on the message. The amended suggestion would be:
```
Warning! Deactivating a device will incur a fee equivalent to the normal
monthly cost for the device (e.g. $1 for an Essentials device); device
fees will not be charged again until the device comes back online.
```

Reply:
Wow big discussing on this warning :)
My bits: the warning in this PR is identical with the one in the UI and which was originally suggested by Alison.
I'm up to further improving it, but it might already be good enough and has the bonus of us being consistent with the UI.
See: https://github.com/balena-io/balena-ui/blob/d5cc98107194dea52ddb0a16b687ac7028dda4ad/src/locales/en/translation.json#L488

If we insist on mentioning that the device has to be offline, then it's probably better to mention that in the action's description like: `Deactivate an offline device.`